packages/camera/camera_avfoundation/ios/camera_avfoundation/Sources/camera_avfoundation/DefaultCamera.swift

```diff
@@ -313,6 +317,90 @@ final class DefaultCamera: FLTCam, Camera {
     }
   }
 
+  func captureToFile(completion: @escaping (String?, FlutterError?) -> Void) {
+    var settings = AVCapturePhotoSettings()
+
+    if mediaSettings.resolutionPreset == .max {
+      settings.isHighResolutionPhotoEnabled = true
+    }
+
+    var fileExtension: String
+
+    let isHEVCCodecAvailable = capturePhotoOutput.availablePhotoCodecTypes.contains(
+      AVVideoCodecType.hevc)
+
+    if fileFormat == .heif, isHEVCCodecAvailable {
+      settings = AVCapturePhotoSettings(format: [AVVideoCodecKey: AVVideoCodecType.hevc])
+      fileExtension = "heif"
+    } else {
+      fileExtension = "jpg"
+    }
+
+    if flashMode != .torch {
+      settings.flashMode = FCPGetAVCaptureFlashModeForPigeonFlashMode(flashMode)
+    }
+
+    let path: String
+    do {
+      path = try getTemporaryFilePath(
+        withExtension: fileExtension,
+        subfolder: "pictures",
+        prefix: "CAP_")
+    } catch let error as NSError {
+      completion(nil, DefaultCamera.flutterErrorFromNSError(error))
+      return
+    }
+
+    let savePhotoDelegate = FLTSavePhotoDelegate(
+      path: path,
+      ioQueue: photoIOQueue,
+      completionHandler: { [weak self] path, error in
+        guard let strongSelf = self else { return }
+
+        strongSelf.captureSessionQueue.async { [weak self] in
+          self?.inProgressSavePhotoDelegates.removeObject(
+            forKey: settings.uniqueID)
+        }
+
+        if let error = error {
+          completion(nil, DefaultCamera.flutterErrorFromNSError(error as NSError))
+        } else {
+          assert(path != nil, "Path must not be nil if no error.")
+          completion(path, nil)
+        }
+      }
+    )
+
+    assert(
+      DispatchQueue.getSpecific(key: captureSessionQueueSpecificKey)
+        == captureSessionQueueSpecificValue,
+      "save photo delegate references must be updated on the capture session queue")
+    inProgressSavePhotoDelegates[settings.uniqueID] = savePhotoDelegate
+    capturePhotoOutput.capturePhoto(with: settings, delegate: savePhotoDelegate)
+  }
+
+  private func getTemporaryFilePath(withExtension ext: String, subfolder: String, prefix: String)
+    throws
+    -> String
```

Comment:
the formatter doesn't complain anything here?

Reply:
It doesn't but I will split arguments into separate lines

Reply:
cc @stuartmorgan-g i suspect something weird happening to our swift formatter.

Reply:
> It doesn't but I will split arguments into separate lines

I was not referring to arguments, I was referring to these 2 lines
```
throws
-> Strings
```

Reply:
> cc @stuartmorgan-g i suspect something weird happening to our swift formatter.

There isn't an "our swift formatter" any more; [we just use Xcode's bundled version now](https://github.com/flutter/packages/pull/9460).